estuary/utils/story.py

```diff
@@ -144,6 +145,59 @@ def get_sibling_nodes_count(self, results, reverse=False):
             return correlated_nodes[::-1]
         return correlated_nodes
 
+    def get_wait_times(self, results):
+        """
+        Get the wait time between two different artifacts.
+
+        :param list results: contains inflated results from Neo4j
+        :return: list of wait time tuples in order of the pipeline (oldest to newest)
+        :rtype: list
+        :raises RuntimeError: if results has less than 2 elements
+        """
+        len_story = len(results)
+        if len_story < 2:
+            raise RuntimeError('This function can\'t be called with one or zero elements')
+
+        # Some services do not have a real completion time because they perform a single action
+        # that takes a negligible amount of time
+        completion_times = {
+            'BugzillaBug': 'creation_time',
+            'DistGitCommit': 'commit_date',
+            'Advisory': 'created_at',
+            'ContainerAdvisory': 'created_at',
+            'FreshmakerEvent': 'time_done',
+            'KojiBuild': 'completion_time',
+            'ModuleKojiBuild': 'completion_time',
+            'ContainerKojiBuild': 'completion_time'
+        }
+
+        wait_times = []
+
+        for index in range(len_story - 1):
+            artifact = results[index]
+            next_artifact = results[index + 1]
+            property_name = completion_times[artifact.__label__]
+            if getattr(artifact, property_name) and next_artifact.timeline_timestamp:
+                completion_time = getattr(artifact, property_name)
```

Comment:
What if `completion_time` is `None` if Estuary-Updater hasn't filled in this info yet?